File: src/main/java/tg/Ipnet/efunerailles/Service/FamilleService.java
package tg.Ipnet.efunerailles.Service;

import java.util.List;
import java.util.Optional;

import tg.Ipnet.efunerailles.Entity.Famille;

public interface FamilleService {
	
	 List<Famille> getAllFamilles();
	    Optional<Famille> getFamilleById(Long id);
	    Famille createFamille(Famille famille);
	    Famille updateFamille(Long id, Famille familleDetails);
	    void deleteFamille(Long id);


}

File: frontend/src/app/core/services/famille.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Famille } from '../models/famille.model';

@Injectable({
  providedIn: 'root'
})
export class FamilleService {
  private apiUrl = 'http://localhost:8181/api/familles'; // À confirmer

  constructor(private http: HttpClient) { }

  getAllFamilles(): Observable<Famille[]> {
    return this.http.get<Famille[]>(this.apiUrl);
  }

  getFamilleById(id: number): Observable<Famille> {
    return this.http.get<Famille>(`${this.apiUrl}/${id}`);
  }

  createFamille(famille: Famille): Observable<Famille> {
    return this.http.post<Famille>(this.apiUrl, famille);
  }

  updateFamille(id: number, famille: Famille): Observable<Famille> {
    return this.http.put<Famille>(`${this.apiUrl}/${id}`, famille);
  }

  deleteFamille(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}
